pkg/virt-handler/vm.go

```diff
@@ -955,14 +955,18 @@ func (d *VirtualMachineController) updateInterfacesFromDomain(vmi *v1.VirtualMac
 			// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 			if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
 				newInterface.InterfaceName = interfaceStatus.InterfaceName
+				newInterface.InfoSource = interfaceStatus.InfoSource
 				// Do not update if interface has Masquerede binding
 				// virt-controller should update VMI status interface with Pod IP instead
 				if !isForwardingBindingInterface {
 					newInterface.IP = interfaceStatus.Ip
 					newInterface.IPs = interfaceStatus.IPs
 				}
 				delete(domainInterfaceStatusByMac, interfaceMAC)
+			} else {
+				newInterface.InfoSource = netvmispec.InfoSourceDomain
```

Comment:
This will work for regular interfaces but not for SR-IOV interfaces.
Can you please add a check on the SR-IOV e2e tests with the expected value?

Reply:
> This will work for regular interfaces but not for SR-IOV interfaces.

for the sriov it will add the interfaces from 
https://github.com/kubevirt/kubevirt/pull/6793/files#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR984
because its a hosdev and not in the `domain.Spec.Devices.Interfaces`

Reply:
What about cases where you have no guest-agent? How will we see the SR-IOV interface?
Can it be that we do not report it today at all if no guest-agent is present? It doesn't seem right, I think it is reported through the pod interfaces. Line 866 above.
This requires checking/testing.

Reply:
we don't report them if there is no guest-agent, its not new
you can also see that before the first guest agent report we don't see them

Lets say this is the yaml interfaces / network
```
      interfaces:
      - masquerade: {}
        name: default
      - macAddress: 02:00:00:e1:62:a2
        name: sriov3
        sriov: {}


  networks:
  - name: default
    pod: {}
  - multus:
      networkName: sriov
    name: sriov3
```

Before first guest agent report, only the pod network is added via line 866 (as it looks on the interfaces from the pod)
```
  interfaces:
  - infoSource: domain
    ipAddress: 10.244.1.8
    ipAddresses:
    - 10.244.1.8
    mac: 52:54:00:bc:ad:c6
    name: default
```

After the guest agent report, we see all the info,
we can still see that sriov is only from the guest-agent according infoSource,
this is why there is the code block
https://github.com/kubevirt/kubevirt/pull/6793/files#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR973-R984
which adds all the interfaces that were not handled before in this function (hostdev that are not in the domain.spec such as sriov, manually added interfaces) 
```
  - infoSource: domain, guest-agent
    interfaceName: eth0
    ipAddress: 10.244.1.8
    ipAddresses:
    - 10.244.1.8
    mac: 52:54:00:bc:ad:c6
    name: default
  - infoSource: guest-agent
    interfaceName: sriov
    ipAddress: 192.168.1.1
    ipAddresses:
    - 192.168.1.1
    - fe80::ff:fee1:62a2
    mac: 02:00:00:e1:62:a2
    name: sriov3
```

Reply:
The above snippets are from a real run (there was a guest agent installed. but before the first report this what was shown).
The question on stack is if `getPodInterfacefromFileCache` have something about the sriov data when there is no guest agent (or when there is unless i am wrong in the observation)
https://github.com/kubevirt/kubevirt/blob/7655a29a3d010234f56c5ae084c13c07b1033eb2/pkg/virt-handler/vm.go#L870
anyhow imho it is not related to this PR, and we can take it offline, unless you think otherwise

Reply:
In the scenario I ran there were 2 interfaces, imo it affects the flow.
The problem is not new, i am aware of it and described it above, but np lets take it to a follow effort please.